hack/ci/e2e-conformance.sh

```diff
@@ -116,6 +121,7 @@ cleanup() {
 
 # our exit handler (trap)
 exit-handler() {
+  unset KUBECONFIG
```

Comment:
KUBECONFIG is being set during `run_tests`, so this is to ensure that we aren't attempting to connect to the workload cluster when we should be connecting to the kind cluster.